texttospeech/api/TextToSpeech/TextToSpeech.cs

```diff
@@ -0,0 +1,288 @@
+﻿/*
+ * Copyright (c) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using CommandLine;
+using System;
+using System.IO;
+using Google.Cloud.TextToSpeech.V1;
+
+namespace GoogleCloudSamples
+{
+
+    [Verb("list", HelpText = "List available voices.")]
+    class ListArgs
+    {}
+
+    [Verb("synthesize", HelpText = "Synthesize input to audio")]
+    class SynthesizeArgs
+    {
+        [Value(0, HelpText = "The text to translate.",
+            Required = true)]
+        public string Text { get; set; }
+
+        [Option('f', HelpText = "Source formatting")]
+        public string SourceFormat { get; set; }
+
+    }
+
+    [Verb("synthesize-file", HelpText = "Synthesize a file to audio")]
+    class SynthesizeFileArgs
+    {
+        [Value(0, HelpText = "The file to synthesize",
+            Required = true)]
+        public string Text { get; set; }
+
+        [Option('f', HelpText = "Source formatting")]
+        public string SourceFormat { get; set; }
+    }
+
+    public class TextToSpeech
+    {
+        public static void Main(string[] args)
+        {
+            Console.OutputEncoding = System.Text.Encoding.Unicode;
+            Parser.Default.ParseArguments<ListArgs, SynthesizeArgs,
+                SynthesizeFileArgs>(args).MapResult(
+                (ListArgs largs) => ListVoices(largs),
```

Comment:
Unless all of the other C# samples use these acronyms/short variables, please use full words

Reply:
How about:

`(ListArgs largs) => ListVoices()`

Reply:
Reference for the library:
https://github.com/commandlineparser/commandline

Reply:
Ah Ha! Here's one that might be useful: https://github.com/GoogleCloudPlatform/dotnet-docs-samples/blob/master/translate/api/Translate/Translate.cs#L89

Reply:
^--- here's how a no-argument static method is being called by CommandLine in another sample

Reply:
Done